### PR TITLE
libafl_libfuzzer: documentation and build script

### PR DESCRIPTION
**Pull request by addisoncrump**
This adds additional usage documentation for `libafl_libfuzzer` for both using with Rust and C targets. Additionally, it provides a build script which assembles the required `libFuzzer.a` for use with C-like targets.

### CODE REVIEW

File: libafl_libfuzzer/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+You will commonly need this for flags such as `-ignore_crashes=1` and `-timeout=5`. In addition
+to partial support of libfuzzer flags, `libafl_libfuzzer` offers:
+
+- `-dedup=n`, with `n` = 1 enabling deduplication of crashes by stacktrace.
+- `-grimoire=n`, with `n` set to 0 or 1 disabling or enabling [grimoire] mutations, respectively.
+    - if not specified explicitly, `libafl_libfuzzer` will "guess" which setting is appropriate
+    - you should disable grimoire if your target is not string-like
+- `-report=n`, with `n` = 1 causing `libafl_libfuzzer` to emit a report on the corpus content.
+- `-skip_tracing=n`, with `n` = 1 causing `libafl_libfuzzer` to disable comparison log tracing.
```
Comment (s1341) [inline libafl_libfuzzer/README.md:116]: 'comparison log', should be `cmplog` no? I don't think anyone knows what 'comparison log' is...

File: libafl_libfuzzer/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+When the runtime is used standalone, flags may be passed just like normal libFuzzer.
+
+You will commonly need this for flags such as `-ignore_crashes=1` and `-timeout=5`. In addition
+to partial support of libfuzzer flags, `libafl_libfuzzer` offers:
+
+- `-dedup=n`, with `n` = 1 enabling deduplication of crashes by stacktrace.
+- `-grimoire=n`, with `n` set to 0 or 1 disabling or enabling [grimoire] mutations, respectively.
+    - if not specified explicitly, `libafl_libfuzzer` will "guess" which setting is appropriate
```
Comment (s1341) [inline libafl_libfuzzer/README.md:113]: "Guess" based on what? corpus content?

**Review by s1341 — APPROVED** [review]
Aside from the two nitpicks, looks good to me...

**Comment by s1341** [comment]
Oh. One last thing. Can we maybe make the build script a `cargo make` toml or a `build.rs`?

**Comment by addisoncrump** [comment]
> Oh. One last thing. Can we maybe make the build script a `cargo make` toml or a `build.rs`?

Potentially yes on the `cargo-make`, but definitely no on `build.rs`. For now, the `.sh` is just easier :slightly_smiling_face:

**Comment by s1341** [comment]
yeah. it's just inconsistent.

**Review by s1341 — APPROVED** [review]
lgtm

**Comment by domenukk** [comment]
cargo make should be trivial, right? Just paste the script in there, basically

**Comment by addisoncrump** [comment]
Ah, yeah, but it wouldn't be windows compat. I think we already have a strategy for this, I'll make a follow-up PR in the oncoming days.

**Pull request merged**